azuredevops/internal/service/build/resource_resource_authorization.go

```diff
@@ -120,55 +154,64 @@ func resourceResourceAuthorizationDelete(d *schema.ResourceData, m interface{})
 	// because the resource to delete might have had this parameter set to true, we overwrite it
 	authorizedResource.Authorized = converter.Bool(false)
 
-	err = sendAuthorizedResourceToAPI(clients, authorizedResource, projectID)
+	err = sendAuthorizedResourceToAPI(clients, authorizedResource, projectID, definitionID)
 	if err != nil {
 		return fmt.Errorf(msgErrorFailedResourceDelete, err)
 	}
 
-	return err
+	return nil
 }
 
 func resourceResourceAuthorizationUpdate(d *schema.ResourceData, m interface{}) error {
 	clients := m.(*client.AggregatedClient)
-	authorizedResource, projectID, err := expandAuthorizedResource(d)
+	authorizedResource, projectID, definitionID, err := expandAuthorizedResource(d)
 	if err != nil {
 		return fmt.Errorf(msgErrorFailedResourceUpdate, err)
 	}
 
-	err = sendAuthorizedResourceToAPI(clients, authorizedResource, projectID)
+	err = sendAuthorizedResourceToAPI(clients, authorizedResource, projectID, definitionID)
 	if err != nil {
 		return fmt.Errorf(msgErrorFailedResourceUpdate, err)
 	}
 
 	return resourceResourceAuthorizationRead(d, m)
 }
 
-func flattenAuthorizedResource(d *schema.ResourceData, authorizedResource *build.DefinitionResourceReference, projectID string) {
+func flattenAuthorizedResource(d *schema.ResourceData, authorizedResource *build.DefinitionResourceReference, projectID string, definitionID int) {
 	d.SetId(*authorizedResource.Id)
-	d.Set("resource_id", *authorizedResource.Id)
-	d.Set("type", *authorizedResource.Type)
-	d.Set("authorized", *authorizedResource.Authorized)
+	d.Set("resource_id", authorizedResource.Id)
+	d.Set("type", authorizedResource.Type)
+	d.Set("authorized", authorizedResource.Authorized)
 	d.Set("project_id", projectID)
+	d.Set("definition_id", definitionID)
 }
 
-func expandAuthorizedResource(d *schema.ResourceData) (*build.DefinitionResourceReference, string, error) {
+func expandAuthorizedResource(d *schema.ResourceData) (*build.DefinitionResourceReference, string, int, error) {
 	resourceRef := build.DefinitionResourceReference{
 		Authorized: converter.Bool(d.Get("authorized").(bool)),
 		Id:         converter.String(d.Get("resource_id").(string)),
 		Name:       nil,
 		Type:       converter.String(d.Get("type").(string)),
 	}
 
-	return &resourceRef, d.Get("project_id").(string), nil
+	return &resourceRef, d.Get("project_id").(string), d.Get("definition_id").(int), nil
 }
 
-func sendAuthorizedResourceToAPI(clients *client.AggregatedClient, resourceRef *build.DefinitionResourceReference, project string) error {
+func sendAuthorizedResourceToAPI(clients *client.AggregatedClient, resourceRef *build.DefinitionResourceReference, projectID string, definitionID int) error {
 	ctx := context.Background()
-
-	_, err := clients.BuildClient.AuthorizeProjectResources(ctx, build.AuthorizeProjectResourcesArgs{
-		Resources: &[]build.DefinitionResourceReference{*resourceRef},
-		Project:   &project,
-	})
+	var err error
+	if definitionID == 0 {
```

Comment:
If `definition_id ` not configured. `nil ` may be a more proper way, `definition_id ` is not configured, should do this operation.

Reply:
Issue here is that if you do a `d.Get("definition_id").(int)` you'll get the `zero` value. The other alternative, as you mentioned, would be using `d.GetOk`, check if `true` return an `*int` or `false` than return `nil` for an `*int`. Because 0 is an invalid value for a definition ID in DevOps, I can live with the implementation here. 

![image](https://user-images.githubusercontent.com/14177833/87674606-846c9b80-c776-11ea-82a5-4cc81a76b43f.png)